[#2536] show skip turn icon when hovering active unit

weiquu opened the pull request:
This fixes issue #2536 

My wallet address is 0x848fBf0588B448f14a2a8BA84Ef12711677073Db

Changes:
- Added `frame.png` and `skip.svg` assets to phaser's autoload
- Added 'no_action' hint type
- In creature's `hint` method, added skip turn frame and icon as sprites if hint type is 'no_action'
- Called `hint` method with appropriate hint type for `onHoverOff` and `onSelectFn` in hexgrid's `queryHexes` method

Screenshot:
![Screenshot 2024-03-05 at 2 54 57 PM](https://github.com/FreezingMoon/AncientBeast/assets/48304907/f7f19bf3-45df-41d3-b4b1-e114d96e1210)

Very new to game dev, so would greatly appreciate any feedback (:

DreadKnight commented:
@weiquu Heya! Did some testing, pretty neat, needs a bit more work though:
- the icon should have semi-transparent black background, like the UI icons
- "Skip Turn" text shows up on previous creatures for no reason
- the text+icon also show up on inactive unit that was hovered
![Screenshot_20240315_163551](https://github.com/FreezingMoon/AncientBeast/assets/587487/73c06dfb-f00a-4dcf-9c44-19a8d9ac1782)

weiquu commented:
Hey @DreadKnight, I've made the following changes:
- Added the semi-transparent black background and increased the size of the icon within the frame (see screenshot below)
- Added checks and timeout for game hints. The reason that the text (and icon) were showing up on inactive units was because I missed out the checks for `activeCreature`. After adding those in, I also ran into an issue with the "Skip turn" text appearing even though the unit shouldn't be active. This was caused by the timeout (`nextCreature()` in `game.ts`) in setting the next active unit. I got around this by setting the "Skipped" text to appear after a slightly longer timeout, although I'm unsure if there's a better way to do this

<img width="139" alt="Screenshot 2024-03-18 at 12 34 52 AM" src="https://github.com/FreezingMoon/AncientBeast/assets/48304907/877d6c61-8ba7-4602-96af-a4b691c21666">

DreadKnight commented:
@weiquu Pretty good.  *Might* make a new issue to sort of avoid showing 2x "Skil turn" texts if hovering on-off-on fast, maybe just having the icon pushed out, not the next as well. Will merge 👍🏻

DreadKnight commented:
@weiquu I've created the issue -> #2562

Also bounty for #2536 has been sent -> https://www.mintme.com/explorer/tx/0xefeabdbe4d13d735d26983b43fc2e8a714d5c9313d44a1a493ff07ddacd0e3fd 🪙

DreadKnight commented:
@weiquu Heya! It seems this kinda broke for some reason; I've opened new issue with bounty 🪙 for it -> #2583
